### PR TITLE
Migrated AdminService to new design

### PR DESCRIPTION
**Pull request by netopyr**
This PR migrates the AdminService to the new design:

- Makes the `preHandle()` method generic. It has only one parameter `context` and is defined in `TransactionHandler`.
- Introduces interfaces for each store (e.g. `ReadableTokenStore`) in the corresponding api-module
- Adds the suffix 'Impl' to names of store implementations (e.g. `ReadableTokenStoreImpl`). The classes implement the new interfaces.
- `PreHandleContext` will become an interface in an upcoming PR. To prepare for this change, tests were updated.
- We will stop to use `AccountAccess` soon. Instead the new interface `ReadableAccountStore` should be used instead.

### CODE REVIEW

File: hedera-node/hedera-admin-service/src/main/java/com/hedera/node/app/service/admin/ReadableSpecialFileStore.java
Comment (hendrikebbers) [inline hedera-node/hedera-admin-service/src/main/java/com/hedera/node/app/service/admin/ReadableSpecialFileStore.java:26]: FUnctionalInterface

**Review by hendrikebbers — APPROVED** [review]

File: hedera-node/hedera-admin-service/src/main/java/com/hedera/node/app/service/admin/ReadableSpecialFileStore.java
Comment (netopyr) [reply hedera-node/hedera-admin-service/src/main/java/com/hedera/node/app/service/admin/ReadableSpecialFileStore.java:26]: Not sure, if that would make sense here. In general, a store contains complex business logic and is not something you want to define as a lambda. I will leave it to the code owners of this service to decide.

Comment (kimbor) [reply hedera-node/hedera-admin-service/src/main/java/com/hedera/node/app/service/admin/ReadableSpecialFileStore.java:26]: This interface is currently located in AdminService but it has been decided to move it to FileService. Once in FileService it will probably grow to have other methods.

**Review by kimbor — APPROVED** [review]

**Pull request merged**
